### PR TITLE
Navigation overlay menu: Fix submenu spacing.

### PR DESCRIPTION
**Pull request by jasmussen**
## Description

In the overlay menu, nested menu items were bunched together in the editor:

before editor

But on the frontend things looked right:

before front

This PR fixes that, now they both look the same:

Screenshot 2021-10-21 at 10 18 32

## How has this been tested?

Insert a menu with nested submenu items. Test the appearance in the overlay menu, editor and front should look the same.

Also test the navigation screen, as this PR removes some CSS that appeared to be dead. It looks right to me.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/navigation-link/editor.scss [hunk 1]
```diff
@@ -1,12 +1,3 @@
-// Normalize Link and edit containers, to look mostly the same.
-.wp-block-navigation__submenu-container {
```
Comment (jasmussen) [inline packages/block-library/src/navigation-link/editor.scss:2]: This doesn't appear to be the same. I think I recall menu item input fields looking gray form fields. This appears to be handled by the input component itself now.

File: packages/block-library/src/navigation-link/editor.scss [hunk 2]
```diff
@@ -67,12 +54,6 @@
 	}
 }
 
-.wp-block-navigation .block-editor-block-list__block[data-type="core/navigation-link"] {
-	& > .block-editor-block-list__insertion-point {
```
Comment (jasmussen) [inline packages/block-library/src/navigation-link/editor.scss:71]: This insertion point is now popovered, so it isn't even in context of the navigation link anymore, so this should be dead CSS.

**Comment by jasmussen** [comment]
Andrew I asked for your review not urgently, but because it replaces a hardcoded gap value in the overlay menu with the new CSS variable. Just wanted to get that sanity checked, thank you!

**Comment by jasmussen** [comment]
Temporarily setting this as in progress, as I need to rebase it off of #35820 and push some small tweaks.

**Comment by jasmussen** [comment]
Alright, I've rebased this one, and simplified the rules a bit following feedback in #35820. Top and bottom margin was applied to all navigation link containers, causing extra spacing between block types in the overlay menu:

Screenshot 2021-10-21 at 12 59 09

Now that spacing is only applied as a top padding on nested items, fixing it:
Screenshot 2021-10-21 at 12 52 02

Here's some test content you can use:

```
<!-- wp:navigation {"overlayMenu":"always"} -->
<!-- wp:navigation-link {"label":"About","type":"page","id":30315,"url":"http://local-wordpress.local/about/","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"New Page","type":"page","id":30646,"url":"http://local-wordpress.local/new-page/","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-submenu {"label":"Journal","type":"page","id":28540,"url":"http://local-wordpress.local/journal/","kind":"post-type","isTopLevelItem":true} -->
<!-- wp:navigation-link {"label":"New Page","type":"page","id":30646,"url":"http://local-wordpress.local/new-page/","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"New Page","type":"page","id":30646,"url":"http://local-wordpress.local/new-page/","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"New Page","type":"page","id":30646,"url":"http://local-wordpress.local/new-page/","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-submenu {"label":"New Page","type":"page","id":30646,"url":"http://local-wordpress.local/new-page/","kind":"post-type","isTopLevelItem":false} -->
<!-- wp:navigation-link {"label":"New Page","type":"page","id":30646,"url":"http://local-wordpress.local/new-page/","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"New Page","type":"page","id":30646,"url":"http://local-wordpress.local/new-page/","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"New Page","type":"page","id":30646,"url":"http://local-wordpress.local/new-page/","kind":"post-type","isTopLevelLink":false} /-->

<!-- wp:navigation-link {"label":"New Page","type":"page","id":30646,"url":"http://local-wordpress.local/new-page/","kind":"post-type","isTopLevelLink":false} /-->
<!-- /wp:navigation-submenu -->
<!-- /wp:navigation-submenu -->

<!-- wp:navigation-link {"label":"Home","type":"page","id":28543,"url":"http://local-wordpress.local/","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:navigation-link {"label":"Contact","type":"page","id":827,"url":"http://local-wordpress.local/contact/","kind":"post-type","isTopLevelLink":true} /-->

<!-- wp:search {"showLabel":false,"buttonUseIcon":true} /-->
<!-- /wp:navigation -->
```

Verify the contents of the overlay menu look good and nicely spaced out. 

The original issue with the editor remains fixed.

**Review by andrewserong — APPROVED** [review]
Thanks for the ping @jasmussen!

The changes to the gap values look good to me, and I tested that the spacing is now consistent in the editor and on the front end (and when changing via the block spacing UI):

| Before | After |
| --- | --- |
| image | image |

If we want to give users control over spacing within a sub-menu further down the track, it should be easy enough to opt into the blockGap feature, too, if we need to.

The CSS removals look good to me, I couldn't see any unexpected differences between trunk and this PR. The submenu container styling appeared to be correct, and the reasoning for removing the insertion point `display:none` sounds reasonable too, given it's in a popover. Since I haven't spent much time looking at how the Navigation block works, feel free to get another review too, if this is an area where you need a confidence check!

Unrelated to this PR, I noticed that if (with overlay switched off) the submenus are set to Open on click, then the position of the icon is the same in the editor and on the front end. However, if it's set to hover, then on the front end the right arrow sits too close to the edge of the div (it looks like the arrow gets rendered outside of the `a` element that sets the padding?):

| Editor | Front end |
| --- | --- |
| image | image |

Also, I did notice that while selecting blocks in the overlay state, the block's toolbar controls are not visible (presumably hidden by the overlay's background?). Is that a known issue? (Obviously not related to this PR as it's happening on trunk, too). I'm testing in TT1-Blocks.

Those two things are outside the scope of changes in this PR, though, so just thought I'd share them since I noticed them, but otherwise this PR LGTM!

**Comment by jasmussen** [comment]
> If we want to give users control over spacing within a sub-menu further down the track, it should be easy enough to opt into the blockGap feature, too, if we need to.

Sounds good. We definitely want some customizability, just have to figure out how far we can go on this one, but good to know!

> Unrelated to this PR, I noticed that if (with overlay switched off) the submenus are set to Open on click, then the position of the icon is the same in the editor and on the front end. However, if it's set to hover, then on the front end the right arrow sits too close to the edge of the div (it looks like the arrow gets rendered outside of the a element that sets the padding?):

Great catch, I think I can fix that up. I'll set a reminder and look at it separately!

> Also, I did notice that while selecting blocks in the overlay state, the block's toolbar controls are not visible (presumably hidden by the overlay's background?). Is that a known issue? (Obviously not related to this PR as it's happening on trunk, too). I'm testing in TT1-Blocks.

Yes, this is intentional _for now_ insofar as you can't actually edit items inside the overlay state very well. It's definitely something to be revisited, but it all relates to how much or how little control we want to/can add here. 

Thanks for the review! I'm feeling pretty good about this one, so I'm going to land it when the checks pass, but I'm always easy to find for followups if there are any!

**Pull request merged**